### PR TITLE
Add agent-eval-harness evaluation for rfe.speedrun

### PR DESCRIPTION
**Pull request by astefanutti**
## Summary

Add evaluation configuration and dataset for the `rfe.speedrun` pipeline using the [agent-eval-harness](https://github.com/opendatahub-io/agent-eval-harness).

### eval.yaml — evaluation config
- **7 judges**: 5 inline checks (files_exist, frontmatter_valid, run_report_exists, recommendation_consistency, pipeline_flow) + 2 LLM judges (rfe_quality, revision_quality)
- **Pairwise A/B comparison** for cross-run/cross-model benchmarking (e.g., Opus vs Sonnet)
- **Tool interception**: auto-answer AskUserQuestion from test case context, block Jira calls during eval
- **Batch mode** with `batch_pattern: "RFE-{n:03d}"` for per-case artifact mapping

### eval/config — judge prompt files
- `pairwise-judge.md`: blind A/B comparison with RFE-specific criteria

## How to run

```bash
# Setup (one-time)
/eval-setup

# Analyze skill and generate eval.yaml (already done)
/eval-analyze --skill rfe.speedrun

# Run evaluation
/eval-run --model opus

# Compare models
/eval-run --model sonnet --subagent-model sonnet --baseline <opus-run-id>
```

## Results from eval runs

| Judge | Opus | Sonnet |
|-------|-----------|--------|
| files_exist | 100% | 95% |
| frontmatter_valid | 100% | 95% |
| run_report_exists | 100% | 100% |
| recommendation_consistency | 100% | 20% |
| pipeline_flow | 100% | 100% |
| rfe_quality | 4.25/5 | 3.95/5 |
| revision_quality | 4.00/5 | 3.00/5 |
| pairwise (Sonnet vs Opus) | — | 8W/7L/5T |

Sonnet produces comparable RFE content (pairwise nearly tied) but has systematic calibration issues (recommendation_consistency 20%).

## Test plan

- [x] Opus evaluation — all 7 judges pass
- [x] Sonnet evaluation — identifies model-specific issues
- [x] Pairwise comparison — meaningful A/B results
- [ ] Verify dataset cases cover edge cases (prescriptive input, vague prompts, oversized scope)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive evaluation framework docs and README with run/interpretation guides.
  * Added evaluation configuration and inline/LLM judge checks for end-to-end validation.
  * Tightened RFE feasibility-review guidance for handling architecture context availability.

* **Chores**
  * Added 20+ evaluation cases covering model serving, autoscaling, observability, CLI, MLflow, and platform features.
  * Added evaluation runtime/configuration files and dataset structure.
  * Added an eval dataset policy requiring anonymization of PII before commit.

### CODE REVIEW

**Comment by jwforres** [comment]
@astefanutti here is the Claude review feedback below, a lot of this seems like minor documentation cleanup items and not really blockers to merging,  easy cleanup to have claude go do real fast :) I know we have the outstanding question on the dataset, did you want to merge without the dataset for now?

----

  Issues / Questions                                                                       
   
  1. No dataset included — The PR body says "7 judges" and "20 test cases" but             
  eval/dataset/cases/ is not in the diff. The commit message says "Dataset to be added
  separately after content review." This means the eval config can't actually be run as-is.
   Is there a follow-up PR planned? Consider mentioning this explicitly in the README or
  adding a placeholder directory.
  2. eval.md frontmatter suggested_judges vs eval.yaml judges diverge — eval.md suggests
  review_accuracy as an LLM judge, but eval.yaml implements revision_quality instead. The  
  recommendation_consistency inline check also isn't in the suggested list. This isn't a
  bug (eval.md is just the analysis cache), but worth noting for consistency — either      
  update eval.md to reflect the final judges or add a note that the suggestions were
  refined.
  3. pipeline_flow judge is fragile — It detects phases via string matching on stdout
  ("Phase 1", "auto-fix", "submit"). If skill output format changes (e.g., log messages are
   reworded), this judge silently degrades. The non_fatal_patterns list is also a
  maintenance concern — new legitimate tracebacks could slip through. Consider documenting 
  this fragility or adding a comment in the check.
  4. frontmatter_valid score consistency check — The pass logic check (pass iff score >= 7 
  AND no criterion is 0) computes total from individual scores but compares against        
  fm.get('pass'). If the score field in frontmatter differs from the sum of criterion
  scores (which it can, since score is set independently), the check might pass frontmatter
   that has an inconsistent score field vs actual criterion sum. Should there also be a
  check that fm['score'] == sum(criterion scores)?
  5. 3-hour timeout (runner_options.timeout: 10800) — This is documented as needed for Opus
   with 20 cases + splits, but it's quite long. Is there a way to run a subset of cases for
   faster iteration? The README doesn't mention this.
  6. pairwise judge uses claude-sonnet-4-6 — Is this intentional to avoid having the same  
  model judge its own output? Worth a brief comment explaining the choice, since someone   
  might think it should match the eval model.
  7. Minor: eval.yaml line 1 uses the name: rfe-speedrun (hyphen) while everywhere else    
  uses rfe.speedrun (dot). Is this a naming convention of the harness, or should it be     
  consistent?
                                                                                           
  Nits                                                                                     
   
  - eval/README.md references /eval-setup and /eval-run — these appear to be slash commands
   from the eval harness. Might be worth noting they require the harness to be installed.
  - The PR body's test plan has one unchecked item: "Verify dataset cases cover edge cases 
  (prescriptive input, vague prompts, oversized scope)" — is this intentionally deferred?

**Comment by astefanutti** [comment]
@jwforres thanks, I've passed that feedback to Claude and pushed an extra commit with the result :) It confirmed these were minor comments.

I have the dataset anonymized locally, I think it'd be useful to add it there, just waiting for your green light on that.

**Review by jwforres — APPROVED** [review]
LGTM

File: eval/dataset/cases/case-019-add-mlflow-integration-for-experiment-tr/input.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+clarifying_context: 'Customers: Internal ML engineering teams, enterprise customers using the AI training platform (e.g.,
+  Ledger AI division, BioTech Labs data science group, RoboDrive ML team)
```
[the remark this answers is not in the document]
Comment (astefanutti) [reply eval/dataset/cases/case-019-add-mlflow-integration-for-experiment-tr/input.yaml:2]: These are fictional company names.
